### PR TITLE
Added kwargs propagation to capture additional JSON API params

### PR DESCRIPTION
**Pull request by Incand**
Hi,

Thanks for your work with this. It saved a lot of time on my project. I needed additional bindings for some other optional API params, specifically the filter functionality.
So I implemented kwargs propagation in most of the generic and specific methods down to _request which now adds all kwargs to the JSON params. 
Everything should be backwards compatible but unfortunately I didn't have time to write tests.

In short the changes I made were:
- Added kwargs propagation to reflect additional optional API params
  - Please note the kwargs_to_jsons_handler function to offset confusion of python standard lib statements and API params
- Added generic "_clone" and "_move" methods
- Minor bug fixes (self.delete -> self._delete, etc.)
- Made the formatting more consistent

I'm looking forward to hearing from you.

### CODE REVIEW

**Comment by pschmitt** [comment]
Thanks, man.
